pkg/registry/generic/registry/store.go

```diff
@@ -268,6 +306,34 @@ func (e *Store) Update(ctx api.Context, obj runtime.Object) (runtime.Object, boo
 		UIDCopy := meta.UID
 		preconditions = &storage.Preconditions{UID: &UIDCopy}
 	}
+	// check if the update is removing all the object's finalizers. If so,
+	// further checks if the object's DeletionGracePeriodSeconds is 0. If so,
+	// delete the object.
+	existing := e.NewFunc()
+	delete := e.shouldDelete(ctx, key, obj, existing)
+	if delete {
+		out := e.NewFunc()
+		glog.V(6).Infof("going to delete %s from regitry, triggered by update", name)
+		if err := e.Storage.Delete(ctx, key, out, preconditions); err != nil {
```

Comment:
The object is deleted here only if the update request empties the finalizers and the DeletionGracePeriods is already set to 0. 
So if the request just empties the finalizers, and the DeletionGracePeriods is *not* 0, then it will be handled as a regular update. The object will be deleted until someone sends a deletion request with DeletionGracePeriod=0.